### PR TITLE
Added basic hc-run

### PR DESCRIPTION
**Pull request by sphinxc0re**
Introduces a command like:

`hc run -p 3400 -b`

where -p is for the port to run the websocket server on... (--port)
-b is short for build, or --package

This will become much more configurable in the future, this is the bare minimum configuration. 
Default port is 8888

Shows CLI feedback something like:
screen shot 2018-11-27 at 10 59 00 pm

### CODE REVIEW

**Comment by Connoropolous** [comment]
I wonder whether we would want to pull out `hc agent` during this PR too, since it seems it does nothing?

@lucksus @sphinxc0re @zippy @thedavidmeister

**Comment by sphinxc0re** [comment]
@Connoropolous I thought of `hc agent` as what is currently `container`. A "power-user-node".

File: cmd/src/cli/run.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        file: package::DEFAULT_BUNDLE_FILE_NAME.into(),
+        hash: "Qm328wyq38924ybogus".into(),
+    };
+
+    let base_config = Configuration {
+        agents: vec![agent_config],
+        dnas: vec![dna_config],
+        ..Default::default()
+    };
+
```
Comment (lucksus) [inline cmd/src/cli/run.rs]: We need an `InstanceConfiguration` and an `InterfaceConfiguration`, no?

Comment (lucksus) [reply cmd/src/cli/run.rs]: Does this work without??

File: cmd/src/cli/run.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+use cli::{self, package};
+use error::DefaultResult;
+use holochain_container_api::{
+    config::{AgentConfiguration, Configuration, DNAConfiguration},
+    container::Container,
+};
+
+pub fn run(package: bool) -> DefaultResult<()> {
```
Comment (Connoropolous) [inline cmd/src/cli/run.rs]: Can you give us some /// rustdoc comments up here about this?

File: cmd/src/cli/run.rs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    let mut container = Container::with_config(base_config);
+
+    container.start_all_interfaces();
+    container.start_all_instances()?;
+
+    let mut rl = rustyline::Editor::<()>::new();
+
+    loop {
+        let readline = rl.readline(">> ")?;
```
Comment (Connoropolous) [inline cmd/src/cli/run.rs]: What does this part do?

Comment (sphinxc0re) [reply cmd/src/cli/run.rs]: To not have an infinite loop making interaction inpossible, I added a simple readline. It displays a simple shell that you can easily `ctrl-c` out of.

File: cmd/src/cli/run.rs (hunk 2 above)
Comment (sphinxc0re) [reply cmd/src/cli/run.rs]: sure

File: cmd/src/cli/run.rs (hunk 1 above)
Comment (sphinxc0re) [reply cmd/src/cli/run.rs]: 🤷‍♂️ 
Give me some pointers. It's not really easy to create these configurations

File: cmd/src/main.rs [hunk 4]
```diff
@@ -96,21 +89,34 @@ enum Cli {
         )]
         language: String,
     },
+    #[structopt(
+        name = "run",
+        alias = "r",
+        about = "starts a development container with an open websocket"
```
Comment (Connoropolous) [inline cmd/src/main.rs]:
```suggestion
        about = "Starts a development container with an open websocket"
```

File: cmd/src/cli/run.rs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        instances: vec![instance_config],
+        interfaces: vec![interface_config],
+        ..Default::default()
+    };
+
+    let mut container = Container::with_config(base_config);
+
+    container.start_all_interfaces();
+    container.start_all_instances()?;
+
```
Comment (Connoropolous) [inline cmd/src/cli/run.rs]: @sphinxc0re we need just a tiny bit of CLI feedback here...
tell the user the service is running

Comment (Connoropolous) [reply cmd/src/cli/run.rs]: And also tell the user that they can type `exit` to exit

File: cmd/src/cli/run.rs (hunk 1 above)
Comment (lucksus) [reply cmd/src/cli/run.rs]: https://github.com/holochain/holochain-rust/blob/develop/container/example-config/basic.toml

File: cmd/src/main.rs [hunk 6]
```diff
@@ -96,21 +89,40 @@ enum Cli {
...
+    Run {
+        #[structopt(
+            long,
+            short,
+            help = "The port to run the websocket server at"
+        )]
+        port: Option<u16>,
+        #[structopt(
+            long,
+            short = "b",
```
Comment (sphinxc0re) [inline cmd/src/main.rs]: why `b`?

File: cmd/src/cli/run.rs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        file: package::DEFAULT_BUNDLE_FILE_NAME.into(),
+        hash: "Qm328wyq38924ybogus".into(),
+    };
+
+    let instance_config = InstanceConfiguration {
+        id: "test-instance".into(),
+        dna: "hc-run-dna".into(),
+        agent: "hc-run-agent".into(),
+        logger: Default::default(),
+        storage: StorageConfiguration::Memory,
```
Comment (lucksus) [inline cmd/src/cli/run.rs]: I think we said it should be persistent. Then this should be set to file storage.

Comment (Connoropolous) [reply cmd/src/cli/run.rs]: Should it? Dunno, wasn't present to that conversation

**Pull request merged**